### PR TITLE
Reload command

### PR DESCRIPTION
**Pull request by NekoSelena**
Reload command always success.

If one of the reloaded file contain an error, the file is skipped and the last working version of the file is loaded instead.
If no file was loaded before, the file containing the error is skipped without being replaced.

The command is supposed to reload the server without stopping it nor affecting the players except if the whitelist is to be enforced.

Files to be reloaded:
 - config.yml
 - whitelist.txt
 - .minecraft/saves/(world)/datapacks
 - plugins

### CODE REVIEW

File: cubic-server/command_parser/commands/Reload.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    else
+        LINFO("autocomplete reload");
+}
+
+void command_parser::Reload::execute(std::vector<std::string>& args, Player *invoker) const {
+    if (invoker) {
+        return;
+    } else {
+        Server::getInstance()->reload();
+        LINFO("Successfully reloaded loot tables, advancements and functions");
```
Comment (STMiki) [inline cubic-server/command_parser/commands/Reload.cpp:16]: The else is useless

**Review by STMiki — APPROVED** [review]

File: cubic-server/Server.cpp [hunk 2]
```diff
@@ -216,11 +216,56 @@ void Server::_downloadFile(const std::string &url, const std::string &path)
...
+        _config.parse("./config.yml");
+    } catch (const std::exception &e) {
+        LERROR(e.what());
+        return;
+    }
+    _maxPlayer = _config.getMaxPlayers();
+    _motd = _config.getMotd();
+    _whitelistEnabled = _config.getWhitelist();
+    _enforceWhitelist = _config.getEnforceWhitelist();
+    return;
```
Comment (Trompettesib) [inline cubic-server/Server.cpp:232]: I think this return is useless

File: cubic-server/Server.cpp [hunk 3]
```diff
@@ -216,11 +216,56 @@ void Server::_downloadFile(const std::string &url, const std::string &path)
...
+*/
+void Server::_reloadWhitelist() {
+    try {
+        if (_whitelistEnabled) {
+            WhitelistHandling::Whitelist whitelistReloaded = WhitelistHandling::Whitelist();
+            _whitelist = whitelistReloaded;
+        }
+    } catch (const std::exception &e) {
+        LERROR(e.what());
+        return;
```
Comment (Trompettesib) [inline cubic-server/Server.cpp:246]: this return too

File: cubic-server/Server.cpp [hunk 4]
```diff
@@ -216,11 +216,56 @@ void Server::_downloadFile(const std::string &url, const std::string &path)
...
+    }
+    return;
```
Comment (Trompettesib) [inline cubic-server/Server.cpp:248]: this one too

File: cubic-server/Server.cpp [hunk 5]
```diff
@@ -216,11 +216,56 @@ void Server::_downloadFile(const std::string &url, const std::string &path)
...
+/*
+**  Reloads the server. Used in the /reload command.
+**  More details in *Reload.hpp*.
+*/
+void Server::reload() {
+    _reloadConfig();
+    _reloadWhitelist();
+    _enforceWhitelistOnReload();
+    /* Reload level.dat + datapacks + plugins */
+    return;
```
Comment (Trompettesib) [inline cubic-server/Server.cpp:260]: and this one

**Review by STMiki — APPROVED** [review]

**Review by Trompettesib — APPROVED** [review]

**Review by huntears — APPROVED** [review]

**Pull request merged**